ambry-quota/src/main/java/com/github/ambry/quota/storage/AmbryStorageQuotaEnforcer.java

```diff
@@ -69,9 +74,18 @@ public boolean shouldThrottle(short accountId, short containerId, QuotaOperation
             return v;
           }
         });
+    logger.debug("Account id {} container id {} quota {}, existing usage {} new size {}, new usage {}", accountId,
+        containerId, quota == Long.MAX_VALUE ? "MAX_VALUE" : String.valueOf(quota), existingUsage.get(), size,
+        storageUsage.get(String.valueOf(accountId)).get(String.valueOf(containerId)));
+    if (exceedQuota.get()) {
+      metrics.quotaExceededCount.inc();
```

Comment:
Besides metric, you can also log warning level messages with acct id, container id, blob id, etc for easy debug.

Reply:
I am planning on adding a callback method when the quota exceeded. In ambryli callback method, I would probably send a kafka event or somehow send out an alert. But in ambry callback event, I would just log it out. Some what like NotificationSystem.